bean/internal/adapter/handler/paymentmethod/paymentmethod.go

```diff
@@ -3,24 +3,104 @@ package paymentmethod
 import (
 	"fmt"
 	"net/http"
+	"strconv"
+
+	"harvest/bean/internal/entity/model"
+	"harvest/bean/internal/entity/viewmodel"
+
+	"harvest/bean/internal/usecase/paymentmethod"
 
 	"harvest/bean/internal/adapter/interfaces"
 )
 
 type handler struct {
+	paymentMethods paymentmethod.UseCase
+
 	view interfaces.CreatePaymentMethodView
 }
 
-func New(view interfaces.CreatePaymentMethodView) http.Handler {
+func New(
+	pm paymentmethod.UseCase,
+	view interfaces.CreatePaymentMethodView,
+) http.Handler {
 	return &handler{
-		view: view,
+		paymentMethods: pm,
+		view:           view,
 	}
 }
 
 func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	err := h.view.Render(w, nil)
+	if r.Method == http.MethodGet {
+		h.render(w, &viewmodel.CreatePaymentMethodViewData{})
+		return
+	}
+
+	if r.Method == http.MethodPost {
+		h.post(w, r)
+		return
+	}
+
+	w.WriteHeader(http.StatusMethodNotAllowed)
+	fmt.Fprintf(w, "Method not allowed")
+}
+
+func (h *handler) post(w http.ResponseWriter, r *http.Request) {
+	form := getFormData(r)
+
+	_, err := h.paymentMethods.Create(
+		"10000000-0000-0000-0000-000000000001",
+		form.Label,
+		form.Last4,
+		model.PaymentMethodBrand(form.Brand),
+		form.ExpMonth,
+		form.ExpYear,
+	)
+
+	if err != nil {
+		h.render(w, &viewmodel.CreatePaymentMethodViewData{
+			Error: err.Error(),
```

Comment:
I don't think this is ideal, but it's good enough for now

Later on, we want re-usable errors that we can check the type of

Those errors should exist per use case